modules/server/src/main/scala/navigate/server/tcs/TcsBaseControllerEpics.scala

```diff
@@ -469,41 +484,116 @@ class TcsBaseControllerEpics[F[_]: Async: Parallel](
     .mountGuideCommand
     .mode(false)
     .post
-    .verifiedRun(ConnectionTimeout)
+    .verifiedRun(ConnectionTimeout) <*
+    stateRef.get.flatMap { s =>
+      s.oiwfs.period
+        .map(setupOiwfsObserve(_, true))
+        .getOrElse(
+          ApplyCommandResult.Completed.pure[F]
+        )
+    }
 
-  override def oiwfsObserve(exposureTime: TimeSpan, isQL: Boolean): F[ApplyCommandResult] = tcsEpics
-    .startCommand(timeout)
-    .oiWfsCommands
-    .observe
-    .interval(exposureTime.toSeconds.toDouble)
-    .oiWfsCommands
-    .observe
-    .output(isQL.fold("QL", ""))
-    .oiWfsCommands
-    .observe
-    .options(isQL.fold("DHS", "NONE"))
-    .oiWfsCommands
-    .observe
-    .numberOfExposures(-1)
-    .oiWfsCommands
-    .observe
-    .path("")
-    .oiWfsCommands
-    .observe
-    .fileName("")
-    .oiWfsCommands
-    .observe
-    .label("")
-    .post
-    .verifiedRun(ConnectionTimeout)
+  def darkFileName(prefix: String, exposureTime: TimeSpan): String =
```

Comment:
Are we reading some files?

Reply:
No, no. The signal processing of the WFS uses an image file to subtract the sky background. It uses a different file for each exposure time. Here we are just telling it which file to use.

Reply:
So we only tell WFS the file name to use, we are not using it directly

Reply:
Precisely